### PR TITLE
Change CI to use matplotlib-base

### LINKED ISSUE #2540: depend on matplotlib-base instead of matplotlib
PR https://github.com/conda-forge/mdanalysis-feedstock/pull/22 suggests that we should use matplotlib-base as our matplotlib dependency (instead of matplotlib).

~Is this a good idea?~ yes

- [ ] Test if it works. 
- [ ] Update our CI to use matplotlib-base instead of matplotlib.
- [ ] Update dependencies in `setup.py`.

### PR DESCRIPTION
**Pull request by IAlibay**
Fixes #2540 

Changes made in this Pull Request:
 - Changes CI conda install from matplotlib to matplotlib-base to match how we release on conda-forge now.

PR Checklist
------------
 - [x] Tests? N/A
 - [x] Docs? N/A
 - [x] CHANGELOG updated? (unecessary)
 - [x] Issue raised/referenced?

### CODE REVIEW

File: .github/workflows/gh-ci.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ defaults:
     shell: bash -l {0}
 
 env:
-  MDA_CONDA_MIN_DEPS: "pip pytest==6.1.2 mmtf-python biopython networkx cython matplotlib scipy griddataformats hypothesis gsd codecov"
+  MDA_CONDA_MIN_DEPS: "pip pytest==6.1.2 mmtf-python biopython networkx cython matplotlib-base scipy griddataformats hypothesis gsd codecov"
```
Comment (IAlibay) [inline .github/workflows/gh-ci.yaml:17]: Sigh.. now I remember why this won't work. Setup will override this and just attempt to pick up normal matplotlib.

**Comment by IAlibay** [comment]
Converting to draft - I need to look into how to optionally pick up either in setup.py

**Comment by IAlibay** [comment]
Turns out I'm wrong, conda installing `matplotlib-base` makes pip aware of the package as `matplotlib`, so it picks up the right version when you then call setup.py (at least as far as my testing locally did).

Unless anyone has a better understanding of what setuptools is doing exactly, I _think_ this just addresses the issue?

**Review by lilyminium — APPROVED** [review]
``matplotlib-base           3.3.4            py36hd391965_0    conda-forge``

LGTM. Thank you @IAlibay!

**Pull request merged**